datastream-server/src/main/java/com/linkedin/datastream/server/assignment/StickyMulticastStrategy.java

```diff
@@ -128,6 +128,9 @@ public StickyMulticastStrategy(Optional<Integer> maxTasks, Optional<Integer> imb
         allAliveTasks.addAll(foundDatastreamTasks);
 
         if (!foundDatastreamTasks.isEmpty()) {
+          if (foundDatastreamTasks.size() > numTasks) {
```

Comment:
Nit: this check is not really necessary here. `limit()` will work on `foundDatastreamTasks` with `size <= numTasks`

Reply:
Keeping this check, so that we don't stream() and limit() every time and do it when necessary. Also, added a log on Sonam's recommendation, so that we can debug if something goes wrong in partition managed.